### PR TITLE
Add support for NETWORKING_WIFI_ADDRESS and NETWORKING_ETHERNET_ADDRESS

### PR DESCRIPTION
**Pull request by schuyler**
Previously, the `buendia-networking` package was designed in such a way that two network interfaces could be defined (`NETWORKING_ETHERNET_ADDRESS` and `NETWORKING_WIFI_ADDRESS`) but only one IP address (`NETWORKING_IP_ADDRESS`).

This PR refactors the networking configuration script into shell functions -- hopefully, for clarity's sake -- and adds support for separate `NETWORKING_WIFI_ADDRESS` and `NETWORKING_ETHERNET_ADDRESS` configuration settings.

The expectation is that if an interface is _defined_ but no static address is set, then the interface should attempt to acquire an address via DHCP.

The extant `NETWORKING_IP_ADDRESS` setting should be considered **legacy**, but it is still supported -- applied first to the Wi-Fi interface if no address is set for it, and then to the Ethernet interface if none is set for it.

STILL TODO:
- ~Thorough testing on an Intel NUC~
- ~Update site configs~
- ~Documentation update~

### CODE REVIEW

File: packages/buendia-networking/data/usr/share/buendia/config.d/10-networking [hunk 1]
```diff
@@ -12,192 +12,223 @@
...
+    [ -n "$1" ] && ( ip a | grep -q " $1:" )
+}
+
+# Configure an interface in /etc/network/interfaces
+configure_interface () {
+    iface=$1
+    addr=$2
+
+    # Bring down the interface during reconfiguration, but don't fail if we can't.
+    ifdown $iface || true
```
Comment (zestyping) [inline packages/buendia-networking/data/usr/share/buendia/config.d/10-networking]: It's odd that this function brings the interface down but not up.  If it's the caller's responsibility to call `ifup` after, I think it should also be the caller's responsibility to call `ifdown` before.

File: packages/buendia-networking/data/usr/share/buendia/config.d/10-networking [hunk 2]
```diff
@@ -12,192 +12,223 @@
...
-        rsn_pairwise=CCMP
-    " > $hostapd_conf
-
-    # Turn this machine into an access point now, and also on future boots.
-    systemctl enable hostapd
-    systemctl start hostapd
-
-else
-    echo "Joining existing wifi network '$NETWORKING_SSID'."
+# Create a file that records the requested settings in a canonical format.
```
Comment (zestyping) [inline packages/buendia-networking/data/usr/share/buendia/config.d/10-networking]: This comment talks about creating a file, but the file isn't created here.  I think it makes sense to factor out a separate function just for all the tmp_settings stuff (lines 135 to 153).

**Comment by zestyping** [comment]
Thanks!  This looks great — all good with me.  I would merge except that it's still marked "NOMERGE" so feel free to merge whenever you're ready!

File: packages/buendia-networking/data/usr/share/buendia/config.d/10-networking [hunk 3]
```diff
@@ -12,192 +12,223 @@
...
-  MAIN_INTERFACE=$NETWORKING_WIFI_INTERFACE
-  USE_WIFI=1
-fi
+# Return true if a network interface can be found by name
+interface_exists () {
+    [ -n "$1" ] && ( ip a | grep -q " $1:" )
+}
+
+# Configure an interface in /etc/network/interfaces
+configure_interface () {
```
[the remark this answers is not in the document]
Comment (schuyler) [reply packages/buendia-networking/data/usr/share/buendia/config.d/10-networking]: :+1:

File: packages/buendia-networking/data/usr/share/buendia/config.d/10-networking (hunk 1 above)
Comment (schuyler) [reply packages/buendia-networking/data/usr/share/buendia/config.d/10-networking]: Agreed. I factored this out into a pair of separate functions, to be run before and after `/etc/network/interfaces` is modified.

File: packages/buendia-networking/data/usr/share/buendia/config.d/10-networking [hunk 4]
```diff
@@ -12,192 +12,223 @@
...
+
+    # Turn wifi power management off -- this is a must for NUC, as otherwise
+    # network performance suffers badly.
+    echo "wireless-power off" >> /etc/network/interfaces
+
+    # Start the interface
+    ifup $NETWORKING_WIFI_INTERFACE
+}
+
+# Create the tmp_settings file and configure the loopback device
```
[the remark this answers is not in the document]
Comment (schuyler) [reply packages/buendia-networking/data/usr/share/buendia/config.d/10-networking]: Done!

**Comment by schuyler** [comment]
I'm just going to do one more round of testing on the NUC to be sure

**Comment by schuyler** [comment]
I've manually tested static and dynamic address allocation for wired and wireless interfaces on the NUC. I'm satisfied with how this works now, so I'm going to go ahead and merge.

**Pull request merged**